### ISSUE TITLE
Docs: Add documentation for on-attach and on-detach events

### ISSUE THREAD

**Issue by patrick-steele-idem** [marko-js/marko#643]
Marko v4 supports `on-attach` and `on-detach` events for any DOM elements that can be used to animate in and animate out DOM elements from the DOM. You can see it in action here: http://markojs.com/try-online/?file=%2Ftodomvc%2Fcomponents%2Ftodomvc-todo-item%2Fdemo.marko&gist=

Calling `event.preventDefault()` in the on-detach event handler prevents an element from being removed from the DOM. After the animation has been completed you should call `event.detach();` to complete the removal of the DOM element out of the DOM.
